pkg/controller/workspacerouting/solvers/openshift_oauth_solver.go

```diff
@@ -60,41 +53,27 @@ func (s *OpenShiftOAuthSolver) GetSpecObjects(spec v1alpha1.WorkspaceRoutingSpec
 	discoverableServices := getDiscoverableServicesForEndpoints(proxyPorts, workspaceMeta)
 	services := append(proxyServices, discoverableServices...)
 
-	routes, proxyEndpoints, podAdditions := s.getProxyRoutes(proxy, workspaceMeta, portMappings)
-	for machineName, machineEndpoints := range proxyEndpoints {
-		exposedEndpoints[machineName] = append(exposedEndpoints[machineName], machineEndpoints...)
-	}
+	routes, podAdditions := s.getProxyRoutes(proxy, workspaceMeta, portMappings)
 
 	return RoutingObjects{
 		Services:         services,
 		Ingresses:        defaultIngresses,
-		Routes:           routes,
+		Routes:           append(routes, defaultRoutes...),
 		PodAdditions:     podAdditions,
-		ExposedEndpoints: exposedEndpoints,
 	}
 }
 
 func (s *OpenShiftOAuthSolver) getProxyRoutes(
 	endpoints map[string][]v1alpha1.Endpoint,
 	workspaceMeta WorkspaceMetadata,
-	portMappings map[string]proxyEndpoint) ([]routeV1.Route, map[string][]v1alpha1.ExposedEndpoint, *v1alpha1.PodAdditions) {
+	portMappings map[string]proxyEndpoint) ([]routeV1.Route, *v1alpha1.PodAdditions) {
 
 	var routes []routeV1.Route
-	var exposedEndpoints = map[string][]v1alpha1.ExposedEndpoint{}
 	var podAdditions *v1alpha1.PodAdditions
-
-	for machineName, machineEndpoints := range endpoints {
+	for _, machineEndpoints := range endpoints {
 		for _, upstreamEndpoint := range machineEndpoints {
 			proxyEndpoint := portMappings[upstreamEndpoint.Name]
 			endpoint := proxyEndpoint.publicEndpoint
-			targetEndpoint := intstr.FromInt(int(endpoint.Port))
-			endpointName := common.EndpointName(endpoint.Name)
-			hostname := common.EndpointHostname(workspaceMeta.WorkspaceId, endpointName, endpoint.Port, workspaceMeta.IngressGlobalDomain)
-			url := fmt.Sprintf("%s://%s", getSecureProtocol(endpoint.Attributes[v1alpha1.PROTOCOL_ENDPOINT_ATTRIBUTE]), hostname)
-
-			// NOTE: openshift oauth-proxy only supports listening on a single port; as a result, we can't proxy more than
-			// one endpoint or we run into cookie issues (each proxied port gets a container and sets a cookie in the browser
-			// once auth is completed).
 			var tls *routeV1.TLSConfig = nil
 			if endpoint.Attributes[v1alpha1.SECURE_ENDPOINT_ATTRIBUTE] == "true" {
 				if endpoint.Attributes[v1alpha1.TYPE_ENDPOINT_ATTRIBUTE] == "terminal" {
```

Comment:
It seems to be workaround. Recently I've seen an issue on Che Server side where someone asks about an ability to configure InsecureEdgeTerminationPolicy on the endpoint level...
Won't it work if we use like TLSTerminationReencrypt for every endpoint?

Reply:
I'm not sure, the logic here is copied from long ago and I haven't had time to dig into the why of it all. Perhaps input from @davidfestal would be helpful here.

Reply:
BTW it's out of the PR scope. I'll create a separate issue for that

Reply:
@amisevsk FYI here is an issue for supporting tls-termination passtrough in the devfile and probably in the plugin-meta.yaml...